care/users/models.py

```diff
@@ -143,12 +143,16 @@ def has_object_read_permission(self, request):
     @staticmethod
     def has_write_permission(request):
         try:
-            return int(request.data["user_type"]) <= User.TYPE_VALUE_MAP["Volunteer"]
-        except TypeError:
-            return User.TYPE_VALUE_MAP[request.data["user_type"]] <= User.TYPE_VALUE_MAP["Volunteer"]
+            user_type = int(request.data["user_type"])
+
+        except ValueError:
+            user_type = request.data["user_type"]
+
         except KeyError:
             # No user_type passed, the view shall raise a 400
             return True
+        finally:
+            return user_type <= User.TYPE_VALUE_MAP["Volunteer"]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [return statement in finally block may swallow exception](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5305091)